drivers/storage/portworx/component/securitycontextconstraints.go

```diff
@@ -185,95 +185,101 @@ func (s *scc) getSCCs(cluster *opcorev1.StorageCluster) []ocp_secv1.SecurityCont
-			AllowHostPID:             false,
-			AllowHostPorts:           false,
-			AllowPrivilegeEscalation: boolPtr(false),
-			AllowPrivilegedContainer: false,
-			FSGroup: ocp_secv1.FSGroupStrategyOptions{
-				Type: ocp_secv1.FSGroupStrategyMustRunAs,
-			},
-			ReadOnlyRootFilesystem: false,
-			RequiredDropCapabilities: []corev1.Capability{
-				"KILL",
-				"MKNOD",
-				"SETUID",
-				"SETGID",
-			},
-			RunAsUser: ocp_secv1.RunAsUserStrategyOptions{
-				Type: ocp_secv1.RunAsUserStrategyMustRunAsRange,
-			},
-			SELinuxContext: ocp_secv1.SELinuxContextStrategyOptions{
-				Type: ocp_secv1.SELinuxStrategyMustRunAs,
-			},
-			SupplementalGroups: ocp_secv1.SupplementalGroupsStrategyOptions{
-				Type: ocp_secv1.SupplementalGroupsStrategyRunAsAny,
-			},
-			Volumes: []ocp_secv1.FSType{
-				ocp_secv1.FSTypeConfigMap,
-				ocp_secv1.FSTypeDownwardAPI,
-				ocp_secv1.FSTypeEmptyDir,
-				ocp_secv1.FSTypeHostPath,
-				ocp_secv1.FSTypePersistentVolumeClaim,
-				ocp_secv1.FSProjected,
-				ocp_secv1.FSTypeSecret,
-			},
-			Groups: nil,
-			Users: []string{
-				fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, CSIServiceAccountName),
-				fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, LhServiceAccountName),
-				fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, PVCServiceAccountName),
-			},
+		AllowHostDirVolumePlugin: true,
+		AllowHostIPC:             false,
+		AllowHostNetwork:         true,
+		AllowHostPID:             false,
+		AllowHostPorts:           true,
+		AllowPrivilegeEscalation: boolPtr(true),
+		AllowPrivilegedContainer: true,
+		AllowedUnsafeSysctls:     []string{"*"},
+		AllowedCapabilities: []corev1.Capability{
+			ocp_secv1.AllowAllCapabilities,
+		},
+		FSGroup: ocp_secv1.FSGroupStrategyOptions{
+			Type: ocp_secv1.FSGroupStrategyRunAsAny,
+		},
+		Priority:               sccPriority,
+		ReadOnlyRootFilesystem: false,
+		RunAsUser: ocp_secv1.RunAsUserStrategyOptions{
+			Type: ocp_secv1.RunAsUserStrategyRunAsAny,
+		},
+		SELinuxContext: ocp_secv1.SELinuxContextStrategyOptions{
+			Type: ocp_secv1.SELinuxStrategyRunAsAny,
+		},
+		SeccompProfiles: []string{"*"},
+		SupplementalGroups: ocp_secv1.SupplementalGroupsStrategyOptions{
+			Type: ocp_secv1.SupplementalGroupsStrategyRunAsAny,
+		},
+		Volumes: []ocp_secv1.FSType{"*"},
+		Groups:  nil,
+		Users: []string{
+			fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, pxutil.PortworxServiceAccountName(cluster)),
+			fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, CollectorServiceAccountName),
+			fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, ServiceAccountNameTelemetry),
+			fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, "px-node-wiper"),
+			fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, "stork-scheduler"),
 		},
 	}
+
+	if gte_ocp_412, err := pxutil.IsSupportedOCPVersion(s.k8sClient, pxutil.OpenshiftPrometheusSupportedVersion); err == nil {
+		if !gte_ocp_412 {
```

Comment:
Can we override this behavior if prometheus is explicitly enabled using `spec.monitoring.prometheus.enabled`? That way we still leave a backdoor for the px deployed prometheus in case there is a remote possibility that we recommend deploying our own prometheus.

Reply:
Updated : Adding px-prometheus user to Portworx SCC if
- either openshift version is less than 4.12
- or px prometheus is enabled